thetis/tracer_eq.py

```diff
@@ -179,12 +179,14 @@ def residual(self, solution, solution_old, fields, fields_old, bnd_conditions=No
                     else:
                         c_in = solution
                         c_ext, uv_ext, eta_ext = self.get_bnd_functions(c_in, uv, elev, bnd_marker, bnd_conditions)
+                        # add interior tracer flux
```

Comment:
This change makes perfect sense to me, but just out of interest: is there something that spurred this, some instability, conservation properties, etc.?

Reply:
(Same for the change in momentum equation)

Reply:
Yes, exactly, the open boundaries were really unstable with active tracers. This change makes a big difference.